apps/wakunode2/app.nim

```diff
@@ -368,15 +360,10 @@ proc updateEnr(app: var App): AppResult[void] =
   let record = enrConfiguration(app.conf, app.netConf, app.key).valueOr:
     return err("ENR setup failed: " & error)
 
-  if app.conf.clusterId != 0:
-    let tRecord = record.toTyped().valueOr:
-      return err("ENR setup failed: " & $error)
-
-    let relayShard = tRecord.relaySharding().valueOr:
-      return err("ENR setup failed: no sharding info")
-
-    if app.conf.clusterId != relayShard.clusterId:
-      return err("ENR setup failed: cluster id mismatch")
+  if (let typedRecord = record.toTyped(); typedRecord.isOk):
+    if (let relayShard = typedRecord.get().relaySharding(); relayShard.isSome):
+      if relayShard.get().clusterId != app.conf.clusterId:
```

Comment:
Minor nitpick. Also, I was wondering if we could encapsulate this snippet somehow as it appears to be used twice in this PR :)

```suggestion
  if (let typedRecord = record.toTyped(); typedRecord.isOk()):
    if (let relayShard = typedRecord.get().relaySharding(); relayShard.isSome()):
      if relayShard.get().clusterId != app.conf.clusterId:
```